index.js

```diff
@@ -0,0 +1,131 @@
+var canvas = document.getElementById('canvas');
+var context = canvas.getContext('2d');
+
+canvas.height = '512';
+canvas.width = '512';
+
+changeMatrixSize('4x4');
+
+var coef = 4;
+var colorLeft = '#000';
+var colorRight = '#fff';
+var lineWidth = 1;
+
+let colorInputLeft = document.getElementById("colorLeft");
+let colorInputRight = document.getElementById("colorRight");
+colorLeft = colorInputLeft.value;
+colorRight = colorInputRight.value;
+
+colorInputLeft.addEventListener("input", function() {
+    colorLeft = colorInputLeft.value;
+}, false);
+colorInputRight.addEventListener("input", function() {
+    colorRight = colorInputRight.value;
+}, false);
+
+
+
+document.getElementById('matrixSize').addEventListener('change', function () {
+    changeMatrixSize(this.value);
+});
+
+function changeMatrixSize(value) {
+    let matrix;
+    if(value == '512x512'){
+        matrix = new Image();
+        matrix.onload = function() {
+            context.drawImage(matrix, 0, 0 , 512 , 512);
+            coef = 512;
+        };
+        matrix.src = "src/512x512.png";
+    }
+    else{
+        matrix = getJson(value);
+        drawFromFile(matrix);
+    }
+}
+
+document.getElementById('lineWidth').addEventListener('change', function () {
+    lineWidth = this.value;
+});
+
+
+function drawFromFile(matrix) {
+    if(matrix!=undefined){
+        let colorDraw;
+        let getColor;
+        let i = 0;
```

Comment:
For now its fine. But it is a bad practise so in the feature declare index variables only in a loop declaration